core/src/main/java/org/apache/carbondata/core/datastore/TableSegmentUniqueIdentifier.java

```diff
@@ -34,6 +36,8 @@
    */
   private AbsoluteTableIdentifier absoluteTableIdentifier;
 
+  private TableProvider tableProvider;
```

Comment:
Keep table provider out of TableSegmentUniqueIdentifier, logically it does not belong here